cmd/kubeadm/app/constants/constants_test.go

```diff
@@ -225,3 +225,68 @@ func TestGetKubeDNSVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestGetKubernetesServiceCIDR(t *testing.T) {
+	var tests = []struct {
+		svcSubnetList string
+		isDualStack   bool
+		expected      string
+		expectedError error
+	}{
+		{
+			svcSubnetList: "192.168.10.0/24",
+			isDualStack:   false,
+			expected:      "192.168.10.0/24",
+			expectedError: nil,
+		},
+		{
+			svcSubnetList: "fd03::/112",
+			isDualStack:   false,
+			expected:      "fd03::/112",
+			expectedError: nil,
+		},
+		{
+			svcSubnetList: "192.168.10.0/24,fd03::/112",
+			isDualStack:   true,
+			expected:      "192.168.10.0/24",
+			expectedError: nil,
+		},
+		{
+			svcSubnetList: "fd03::/112,192.168.10.0/24",
+			isDualStack:   true,
+			expected:      "fd03::/112",
+			expectedError: nil,
+		},
+		{
+			svcSubnetList: "192.168.10.0/24,fd03:x::/112",
+			isDualStack:   true,
+			expected:      "",
+			expectedError: errors.New("unable to parse ServiceSubnet 192.168.10.0/24,fd03:x::/112: failed to parse cidr value:\"fd03:x::/112\" with error:invalid CIDR address: fd03:x::/112"),
+		},
+	}
+
+	for _, rt := range tests {
+		t.Run(string(rt.svcSubnetList), func(t *testing.T) {
+			actual, actualError := GetKubernetesServiceCIDR(rt.svcSubnetList, rt.isDualStack)
+			if actualError != nil {
```

Comment:
might be possible to unify this with the pattern:
```
if (actualError != nil) != rt.expectedError {
```
as long as `rt.expectedError` is a `bool`.